roles/update/tasks/init_monitoring.yml

```diff
@@ -0,0 +1,55 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure update log directory exists.
+  ansible.builtin.file:
+    path: "{{ cifmw_update_artifacts_basedir }}"
+    state: directory
+    mode: "0755"
+
+- name: Create update step monitoring file
+  ansible.builtin.template:
+    src: "update_step.sh.j2"
+    dest: "{{ cifmw_update_artifacts_basedir }}/update_step.sh"
+    mode: "0755"
+
+- name: Create update stage monitoring file
+  ansible.builtin.template:
+    src: "monitor_resources_changes.sh.j2"
+    dest: "{{ cifmw_update_artifacts_basedir }}/monitor_resources_changes.sh"
+    mode: "0755"
+
+- name: Set update step to {{ cifmw_update_state | default("Starting Update") }}
+  ansible.builtin.command:
+    cmd: >
+      {{ cifmw_update_artifacts_basedir }}/update_step.sh
+      {{ cifmw_update_state | default("Starting Update") }}
+
+- name: Initialize monitoring
+  ansible.builtin.command:
+    cmd: "{{ cifmw_update_artifacts_basedir }}/monitor_resources_changes.sh -a init"
+    creates: "{{ cifmw_update_artifacts_basedir }}/update_timeline.log"
+  no_log: "{{ cifmw_nolog | default(true) | bool }}"
+
+- name: Start monitoring
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: >-
+      {{ cifmw_update_artifacts_basedir }}/monitor_resources_changes.sh
+      -a monitor -t {{ cifmw_update_resources_monitoring_interval }} &
```

Comment:
that might not work. 
Try using: `async`

Reply:
Oki, this was a annoying one.

So using `async: 0` means that `pool` becomes relevant.  If too short the parent process get kill and the script as well.  When too long, it should be fine, but it depends on the fact the we stay in the same playbook.

Overall, I change the script to become a daemon, so no need to play with `async` and `poll`.  I though about using `nohup` which should be there, but eventually decided to not intruced the dependency.